standard.md

```diff
@@ -0,0 +1,136 @@
+- **Recommend**: guidelines that all diagnostics are encouraged to meet, but are not mandatory.
+Nevertheless, if a list of compatible diagnostics is compiled, diagnostic scripts meeting
+recommended guidelines will be highlighted.
+- **Suggest**: guidelines that are not required, but are considered to improve the quality of the diagnostics.
+
+Diagnostic scripts are expected to be contributed along with a YAML file containing configuration options.
+This standard defines three different levels of enforceability for the contents of this YAML file:
+
+- **Required**: parameters that must always be present, have a defined meaning and that can only take allowed values.
+- **Reserved**: parameters that can be omitted, but if present, must have a defined meaning and allowed values.
+- **Custom**: extra parameters can be defined as needed by the diagnostics or the tools.
+They must not collide with neither `required` nor `reserved` parameters.
+
+## The diagnostic configuration file
+
+The diagnostic's configuration file *must* consist of YAML file containing key-value mappings for all the parameters needed to run the diagnostic.
+
+### Required options
+
+The following options *must* be present in the configuration file:
+
+- **diagnostic_path** : `str`. Path to the diagnostic executable.
+- **input_files**: `list`. List of absolute paths to the metadata files describing the data to be used by the diagnostic.
+The format of these files is especified in section [The data definition file](the-data-definition-file).
+- **tool**: `str`. Name of the tool used to prepare the data for the diagnostic.
+- **version**: `str` Version of the tool used to prepare the data for the diagnostic.
+- **run_dir**: `str`. Path to the directory to use as current path for the diagnostic execution. *Must* be writable by the diagnostic.
+- **data_dir**: `str`. Path to the directory to store the  diagnostic's generated data on. *Must* be writable by the diagnostic.
+- **plot_dir**: `str`. Path to the directory to store the  diagnostic's generated figures on. *Must* be writable by the diagnostic.
+
+It is *suggested* to provide different directory names for `run_dir`, `data_dir` and `plot_dir`, as it makes the interaction with the results
+easier. However, using the same directory for all purposes is accepted.
+
+### Reserved options
+
+The following options are not required, however if present they must follow the definitions listed below:
+
+- **log_level**: `str`, default value is `info`. Sets the granularity of the diagnostic logs and console output.
+If used, the logger must include `error` `warning`, `info` and `debug` levels .
+- **auxiliary_data_dir** : `str`. Path to the auxiliary directory to store further input data files needed by the diagnostic, such as shapefiles.
+- **max_proc_number** : `integer`, default value is **1**. Maximum number of processors used to run the diagnostic.
```

Comment:
Most diagnostic developers will probably not have the technical skills to implement code that respects this setting. Therefore I'm not sure if this is a good idea because it will result in diagnostics that are broken.

Reply:
It's under this section though:

`The following options are not required, however if present they must follow the definitions listed below`

Reply:
Maybe there are external tools that use multiple preprocessors, so having a name for this option is useful in that case. But anyway if it only adds confusion I can delete it.

Reply:
> Most diagnostic developers will probably not have the technical skills to implement code that respects this setting

The quite easy way to respect the setting is to avoid using more than one processor, which is the default behaviour of most packages that diagnostic developers may use.

Reply:
Not the packages ESMValTool diagnostic developers use. Our Python diagnostic developers mostly use iris or xarray, both of which are based on dask which uses multiple processors by default. There are also other libraries that will use multiple processors by default. We also have several diagnostics written in R, some of the dependencies of these also use multiple processors by default.

Reply:
FIne. Let us forget about max_proc_number